src/base/utils/foreignapps.cpp

```diff
@@ -69,26 +77,181 @@ namespace
+        LPWSTR lpData = new WCHAR[cBuffer];
+        LONG res = ::RegQueryValueExW(handle, lpValueName, NULL, &type, (LPBYTE)lpData, &cbData);
+        if (lpValueName)
+            delete[] lpValueName;
+
+        if (res == ERROR_SUCCESS) {
+            lpData[cBuffer - 1] = 0;
+            result = QString::fromWCharArray(lpData);
+        }
+        delete[] lpData;
+
+        return result;
+    }
+
+    QString searchReg(const REG_SEARCH_TYPE type)
+    {
+        HKEY hkRoot;
+        if (type == USER)
+            hkRoot = HKEY_CURRENT_USER;
+        else
+            hkRoot = HKEY_LOCAL_MACHINE;
+
+        REGSAM samDesired = KEY_READ;
+        if (type == SYSTEM_32BIT)
+            samDesired |= KEY_WOW64_32KEY;
+        else if (type == SYSTEM_64BIT)
+            samDesired |= KEY_WOW64_64KEY;
+
+        QString path;
+        LONG res = 0;
+        HKEY hkPythonCore;
+        res = ::RegOpenKeyExW(hkRoot, L"SOFTWARE\\Python\\PythonCore", 0, samDesired, &hkPythonCore);
+
+        if (res == ERROR_SUCCESS) {
+            QStringList versions = getRegSubkeys(hkPythonCore);
+            qDebug("Python versions nb: %d", versions.size());
+            versions.sort();
+
+            bool found = false;
+            while (!found && !versions.empty()) {
+                const QString version = versions.takeLast() + "\\InstallPath";
+                LPWSTR lpSubkey = new WCHAR[version.size() + 1];
+                version.toWCharArray(lpSubkey);
+                lpSubkey[version.size()] = 0;
+
+                HKEY hkInstallPath;
+                res = ::RegOpenKeyExW(hkPythonCore, lpSubkey, 0, samDesired, &hkInstallPath);
+                delete[] lpSubkey;
+
+                if (res == ERROR_SUCCESS) {
+                    qDebug("Detected possible Python v%s location", qUtf8Printable(version));
+                    path = getRegValue(hkInstallPath);
+                    ::RegCloseKey(hkInstallPath);
+
+                    if (!path.isEmpty() && QDir(path).exists("python.exe")) {
+                        found = true;
+                        path = QDir(path).filePath("python.exe");
+                    }
+                }
+            }
+
+            if (!found)
+                path = QString();
+
+            ::RegCloseKey(hkPythonCore);
+        }
+
+        return path;
+    }
+
+    QString findPythonPath()
+    {
+        QString path = searchReg(USER);
+        if (!path.isEmpty())
+            return path;
+
+        path = searchReg(SYSTEM_32BIT);
+        if (!path.isEmpty())
+            return path;
+
+        path = searchReg(SYSTEM_64BIT);
+        if (!path.isEmpty())
+            return path;
+
+        // Fallback: Detect python from default locations
+        const QFileInfoList dirs = QDir("C:/").entryInfoList({"Python*"}, QDir::Dirs, (QDir::Name | QDir::Reversed));
+        for (const QFileInfo &info : dirs) {
+            const QString path {info.absolutePath() + "python.exe"};
```

Comment:
Does `QFileInfo::absolutePath()` end with separator?

Reply:
No, will fix it.